test/rules_test.js

```diff
@@ -206,6 +206,19 @@ describe('rules', () => {
         ['2x + 7y + 5 + 3y + 9x + 11', '(2 x + 9 x) + (7 y + 3 y) + (5 + 11)'],
     ])
 
+    // ADDING POLYNOMIALS
+
+    suite('add polynomials', rules.ADD_POLYNOMIAL_TERMS, [
+        ['2x + 2x + 2 + 4', '4 x + (2 + 4)'],
+        ['3y^2 - 2y^2 + y^4', '1 y^2 + 1 y^4'],
+        ['x - x', '0 x'],
+        ['2x + 3x + 2y + 3y', '5 x + 5 y'],
+        ['-2y + 3y', '1 y'],
+        ['3 xy + 2 xy', '5 xy'],
+        ['3 xy - 2 xy + x^2y^2', '1 (x^2 y^2) + 1 xy'],
```

Comment:
Can you change these to use `x y` instead of `xy`?  Also, 
```
x^2y^2 --> 1 (x^2 y^2)
should probably be
```
x^2y^2 --> 1 x^2 y^2
```
This will require calling `flattenOperands` on `term` before returning it from the rewrite function of `ADD_POLYNOMIAL_TERMS`.